resolwe_runtime_utils.py

```diff
@@ -275,12 +288,177 @@ def checkrc(rc, *args):
+                        tar.extractall()
+
+                    os.remove(extracted_name)
+                    extracted_name, _ = os.path.splitext(extracted_name)
+        else:
+            # Directory or several files in archive.
+            if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
+                with tarfile.open(extracted_name + '.tar.gz', 'w:gz') as tar:
+                    for fname in glob.glob(os.path.join(temp_dir, '*')):
+                        tar.add(fname, os.path.basename(fname))
+
+            if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+                for path in os.listdir(temp_dir):
+                    shutil.move(os.path.join(temp_dir, path),
+                                './{}'.format(path))
+
+        shutil.rmtree(temp_dir)
+        return extracted_name + '.gz' if imported_format == ImportedFormat.COMPRESSED else extracted_name
+
+    def importUncompressed():
+        """Import uncompressed file."""
+        if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
+            with open(src, 'rb') as f_in, gzip.open(file_name + '.gz', 'wb') as f_out:
+                shutil.copyfileobj(f_in, f_out, CHUNK_SIZE)
+
+        if imported_format != ImportedFormat.COMPRESSED:  # Extracted file required
+            try:
+                shutil.copyfile(src, file_name)
+            except shutil.SameFileError:
+                pass  # Skip copy of downloaded files
+
+        return file_name + '.gz' if imported_format == ImportedFormat.COMPRESSED else file_name
+
+    # Large file download from Google Drive requires cookie and token.
+    response = None
+    if re.match(r'^https://drive.google.com/[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]$', src):
+        session = requests.Session()
+        response = session.get(src, stream=True)
+
+        token = None
+        for key, value in response.cookies.items():
+            if key.startswith('download_warning'):
+                token = value
+                break
+
+        if token is not None:
+            params = {'confirm': token}
+            response = session.get(src, params=params, stream=True)
+
+    elif re.match(r'^(https?|ftp)://[-A-Za-z0-9\+&@#/%?=~_|!:,.;]*[-A-Za-z0-9\+&@#/%=~_|]$', src):
+        response = requests.get(src, stream=True)
+
+    if response:
+        with open(file_name, 'wb') as f:
+            total = response.headers.get('content-length')
+            total = float(total) if total else None
+            downloaded = 0
+            current_progress = 0
+            for content in response.iter_content(chunk_size=CHUNK_SIZE):
+                f.write(content)
+
+                if total is not None and max_progress is not None:
+                    downloaded += len(content)
+                    next_progress = round(
+                        max_progress * total / downloaded, 2)
```

Comment:
This is going to work a bit funny because we don't have `min_progress`. Consider the case where the progress just before running `import_file` is not 0, e.g. if we import multiple files or do some processing prior to running `import_file`.

Reply:
Changed to `import_from`, `import_to`. Please review.